p42/src/bank.rs

```diff
@@ -0,0 +1,109 @@
+use std::collections::HashMap;
+pub enum Error {
+    SenderNotFound,
+    ReceiverNotFound,
+    AttemptedAmountTooLarge,
+    SenderOverflow,
+    ReceiverOverflow,
+    SenderCreditLimitOverflow,
+    SenderCreditLimitTooLarge,
+}
+pub struct User {
+    credit_line: u64,
+    balance: i64,
+}
+pub struct Bank {
+    users: HashMap<String, User>,
+    name: String,
+    credit_interest: u64,
+    debit_interest: u64,
+}
+
+impl Bank {
+    pub fn calc_balance(&self) -> (u64, u64) {
+        let mut liab: u64 = 0;
+        let mut ass: u64 = 0;
+        for (_key, value) in self.users.iter() {
+            if value.balance < 0 {
+                ass += value.balance.unsigned_abs();
+            } else {
+                liab += value.balance.unsigned_abs();
+            }
+        }
+        (liab, ass)
+    }
+
+    pub fn accrue_interest(&mut self) {
+        for (_key, value) in self.users.iter_mut() {
+            if value.balance > 0 {
+                value.balance +=
+                    i64::try_from((value.balance.unsigned_abs() * self.debit_interest) / 10000)
+                        .unwrap();
+            } else {
+                value.balance -=
+                    i64::try_from((value.balance.unsigned_abs() * self.credit_interest) / 10000)
+                        .unwrap();
+            }
+        }
+    }
+
+    pub fn merge_bank(&mut self, new: Bank) {
+        if self.name == new.name {
+            return;
+        }
+        for (key_i, value_i) in new.users {
+            match self.users.get_mut(&key_i) {
+                Some(v) => {
+                    v.balance += value_i.balance;
+                }
+                None => {
+                    self.users.insert(key_i, value_i);
+                }
+            };
+        }
+    }
+
+    pub fn transfer_funds(
+        &mut self,
+        user_send: String,
+        user_rec: String,
+        amount: u64,
+    ) -> Result<(), Error> {
+        let mut temp: &mut User;
+        let amount_i64 = i64::try_from(amount).map_err(|_| Error::AttemptedAmountTooLarge)?;
+
+        match self.users.get(&user_send) {
```

Comment:
You can use `let user_send_ref = self.users.get(&user_send).ok_or(Error)?;` then you can use a similar `get_mut` for other user.